app/services/school_data_importer.rb

```diff
@@ -26,40 +27,29 @@ def run
 
 private
 
-  def gias_schools_csv_url
-    date_string = Time.zone.now.strftime("%Y%m%d")
-    "#{Rails.application.config.gias_api_root}/edubase/downloads/public/edubasealldata#{date_string}.csv"
-  end
-
-  def schools_data_file
-    FileDownload.new(gias_schools_csv_url).fetch
+  def school_data_files
+    @gias_api_client ||= GiasApiClient.new
+    @gias_files ||= @gias_api_client.get_files
+    @gias_files
   end
 
   def row_to_school(row)
     school = School.find_or_initialize_by(urn: row.fetch("URN"))
     school.name = row.fetch("EstablishmentName")
     school.school_type_code = row.fetch("TypeOfEstablishment (code)")
     school.school_type_name = row.fetch("TypeOfEstablishment (name)")
-    school.capacity = row.fetch("SchoolCapacity").to_i
     school.address_line1 = row.fetch("Street")
     school.address_line2 = row.fetch("Locality")
     school.address_line3 = row.fetch("Town")
-    school.address_line4 = row.fetch("County (name)")
-    school.country = row.fetch("Country (name)")
     school.postcode = row.fetch("Postcode")
     school.ukprn = row.fetch("UKPRN")
-    school.previous_school_urn = row.fetch("PreviousEstablishmentNumber")
```

Comment:
oof, scary

without this data, are we gonna have troubles figuring out which school is which after a change?

Reply:
That's what the links file is for 🙂